src/TLuaInterpreter.cpp

```diff
@@ -882,8 +882,8 @@ int TLuaInterpreter::selectCaptureGroup( lua_State * L )
     if( luaNumOfMatch < static_cast<int>(pHost->getLuaInterpreter()->mCaptureGroupList.size()) )
     {
         TLuaInterpreter * pL = pHost->getLuaInterpreter();
-        std::list<std::string>::iterator its = pL->mCaptureGroupList.begin();
-        std::list<int>::iterator iti = pL->mCaptureGroupPosList.begin();
+        auto its = pL->mCaptureGroupList.begin();
+        auto iti = pL->mCaptureGroupPosList.begin();
```

Comment:
Actually can not the iterator declaration moved inside the opening parenthesis of the `for (`